GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -171,6 +178,65 @@ private void ReloadCommitInfo()
 
             if (AppSettings.CommitInfoShowContainedInTags)
                 ThreadPool.QueueUserWorkItem(_ => loadTagInfo(_revision.Guid));
+
+
+            commitSignPicture.Visible = tagSignPicture.Visible = AppSettings.ShowGpgInformation.ValueOrDefault;
+
+            if (AppSettings.ShowGpgInformation.ValueOrDefault)
+            {
+                /* Setup GpgController */
+                _gpgController = new GitGpgController(Module, _revision);
+
+                /* COMMIT section */
+                tagSignPicture.Visible = false;
```

Comment:
this isn't correct - I still can see flicker of the image.
you need to hide the image and set to null in `ResetTextAndImage`:

```
        private void ResetTextAndImage()
        {
            // ...
 
            tagSignPicture.Visible = false;
            tagSignPicture.Image = null;
            commitSignPicture.Visible = false;
            commitSignPicture.Image = null;
        }
```

Also my async implementation isn't perfect - I missed that the image isn't needed to be awaited, it just needs to be loaded asynchronously.
Here's fix from @jbialobr - https://github.com/RussKie/gitextensions/pull/5/commits/9f255dd5d44f936ed9efdd8b2d44bb5fef4cf1b9

Reply:
done

Reply:
The async part is not crrected.

Reply:
@jbialobr what's wrong on the async part ? could you help me understand ?

Reply:
there is no need to wait for async calls to end in `ReloadCommitInfo`, they are essentially "fire and forget" calls.
have a look at  RussKie/gitextensions@9f255dd

Reply:
do you mean that `GetRevisionCommitSignatureStatusAsync` and `GetRevisionTagSignatureStatusAsync` should NOT be async ?

Reply:
ok, let me know how they should be called, thanks.